File: api/Adapters/Driving/Apis/ProjectK.Api/Dtos/v1/Monthly/Responses/MonthlyPlannedTransactionDto.cs
using ProjectK.Core.Enums;

namespace ProjectK.Api.Dtos.v1.Monthly.Responses;

public record MonthlyPlannedTransactionDto(
    Ulid Id,
    string Description,
    decimal Amount,
    AmountType AmountType,
    TransactionType Type,
    Recurrence Recurrence,
    DateOnly ReferringDate,
    DateOnly StartsAt,
    DateOnly? EndsAt,
    Ulid? CategoryId,
    Ulid? PaidTransactionId,
    bool Paid,
    DateOnly? PaidAt,
    decimal? PaidAmount,
    string[] Tags,
    bool Ignore
);

File: app/src/types/dtos/monthly-planned-transaction-dto.ts
import { AmountType, Recurrence, TransactionType } from '..';

export type MonthlyPlannedTransactionDto = {
  id: string;
  description: string;
  amount: number;
  amountType: AmountType;
  type: TransactionType;
  recurrence: Recurrence;
  referringDate: string;
  startsAt: string;
  endsAt?: string;
  categoryId?: string;
  paidTransactionId?: string;
  paid: boolean;
  paidAt?: string;
  paidAmount: number;
  tags: string[];
  ignore: boolean;

  // For optimistic updates
  isChanging?: boolean;
};
